### DISCUSSION TITLE
xmake build targetname时link问题

### DISCUSSION THREAD

**Discussion by DanielKui** [xmake-io/xmake#3967] [Q&A] [answered]
![image](https://github.com/xmake-io/xmake/assets/578118/c89a30dc-b370-40f6-a258-b2fe42e54447)

Hi , 我用 xmake  build -v HXQuoteApi来编译一个target,  但提示link失败了，如果我改为手动执行提示错误的地方，是可以成功link的，
还有就是我使用xmake generator生成makefile后，link也没问题，似乎是xmake build行为跟makefile不一致吗？

**Comment by waruqi** [comment, marked as answer]
一种可能就是 你加了各种脚本，里面 os.cd 把当前路径改乱了，自己看下。。os.cd 全部去掉

**Comment by DanielKui** [comment]
感谢耐心解答
os.ln 是否支持带参数的呢？
比如我想ln -sf A.so.23  A.so

**Reply by waruqi** [reply]
只支持源。目路径，无其他参数
